mqtt-impl/src/main/java/io/streamnative/pulsar/handlers/mqtt/MQTTSubscriptionManager.java

```diff
@@ -31,8 +35,30 @@
 
     private ConcurrentMap<String, List<MqttTopicSubscription>> subscriptions = new ConcurrentHashMap<>(2048);
 
-    public void addSubscriptions(String clientId, List<MqttTopicSubscription> topicSubscriptions) {
-        this.subscriptions.computeIfAbsent(clientId, k -> new ArrayList<>()).addAll(topicSubscriptions);
+    public boolean addSubscriptions(String clientId, List<MqttTopicSubscription> topicSubscriptions) {
+        final AtomicBoolean duplicated = new AtomicBoolean(false);
```

Comment:
Why use ``AtomicBoolean``? It doesn't look like a shared variable.

Reply:
Because this variable is in the lambda expression.